File: frontend/src/app/shared/item/item.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Item} from "../../model/item.model";

const httpOptions = {
  headers: new HttpHeaders({
    'Content-Type':  'application/json'
  })};

@Injectable()
export class ItemService {

  constructor(private  http: HttpClient) {
  }

  getAll(){
    return this.http.get<Item[]>('http://localhost:8080/items');
  }

  saveAll(items: any[]){
    let body = JSON.stringify(items);
    this.http.put('http://localhost:8080/save-items',body,httpOptions).subscribe();

  }


}

File: backend/src/main/java/psi/service/ItemService.java
package psi.service;

import psi.model.Item;

import java.util.List;

public interface ItemService {

    void save(Item item);

    void delete(String id);

    List<Item> findAll();

    List<Item> saveAll(List<Item> items);

    Item getItemById(String id);

}
